hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -130,8 +151,14 @@ public HoodieData<WriteStatus> compact(
     TaskContextSupplier taskContextSupplier = table.getTaskContextSupplier();
     // if this is a MDT, set up the instant range of log reader just like regular MDT snapshot reader.
     Option<InstantRange> instantRange = CompactHelpers.getInstance().getInstantRange(metaClient);
+
+    // Broadcast necessary information.
+    Option<CompactorBroadcastManager> broadcastManagerOpt = getCompactorBroadcastManager(context);
```

Comment:
I would say, we should not even instantiate CompactorBroadcastManager if partial update is not enabled. 

also, broadcastManager naming does not sit well. for eg, why would this be a required entity for non partial update flows.

So, can we rename it to be little generic. 

something like PartialUpdateReaderContext or PartialUpdateContext.

Reply:
All it does is to enable fg reader, and partial update is just a part of it. So emphasizing partial update is too specific.

Reply:
Like you said, partial update check can go before it to avoid broadcast as of now.

Reply:
> and partial update is just a part of it. So emphasizing partial update is too specific.
+1